neqo-http3/src/features/extended_connect/webtransport_session.rs

```diff
@@ -415,7 +415,7 @@ impl WebTransportSession {
             let mut dgram_data = Encoder::default();
             dgram_data.encode_varint(self.session_id.as_u64() / 4);
             dgram_data.encode(buf);
```

Comment:
A second re-allocation of the entire Quic Datagram is happening here, just to prefix it with a varint.

I would assume there is a better way to do that, though I suggest not in this pull request.